pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,276 @@ type DiscoveryReconciler struct {
+				}
+			}
+			i++
+			first = false
+		}
+
+		err = r.Create(context.Background(), peeringCandidate)
+		if err != nil {
+			klog.Errorf("Error when creating Peering Candidate: %s", err)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		} else {
+			// Update the Discovery with the PeeringCandidate
+			discovery.Spec.PeeringCandidate.Name = peeringCandidate.Name
+			discovery.Spec.PeeringCandidate.Namespace = DEFAULT_NAMESPACE
+			if err := r.Update(ctx, &discovery); err != nil {
+				klog.Errorf("Error when updating Discovery %s: %s", discovery.Name, err)
+				return ctrl.Result{}, err
+			}
+			klog.Infof("Discovery %s updated", discovery.Name)
+			setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseSolved)
+		}
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		} else {
+			// Create a Reservation CR
+			reservationCR := createReservationCustomResource(peeringCandidate)
+			err := r.Create(context.Background(), reservationCR)
+			if err != nil {
+				klog.Errorf("Error when creating Reservation: %s", err)
+				setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+			} else {
+				klog.Infof("Reservation %s created", reservationCR.Name)
+			}
+			return ctrl.Result{}, err
+		}
+	case nodecorev1alpha1.PhaseSolved:
+
+		klog.Infof("Discovery %s solved", discovery.Name)
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseSolved)
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	case nodecorev1alpha1.PhaseFailed:
+		klog.Infof("Discovery %s failed, no Peering Candidate found", discovery.Name)
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
 
 	return ctrl.Result{}, nil
 }
 
+// setDiscoveryPhase sets the phase of the discovery
+func setDiscoveryPhase(d *advertisementv1alpha1.Discovery, phase nodecorev1alpha1.Phase) {
+	d.Status.Phase.Phase = phase
+}
+
+// updateSolverStatus updates the status of the discovery
+func (r *DiscoveryReconciler) updateSolverStatus(ctx context.Context, discovery *advertisementv1alpha1.Discovery) error {
+	return r.Status().Update(ctx, discovery)
+}
+
+// getFlavour is a function that returns an array of Flavour that fit the Selector by performing a get request to an http server
+func getFlavour(selector nodecorev1alpha1.FlavourSelector) ([]*nodecorev1alpha1.Flavour, error) {
+	// Marshal the selector into JSON bytes
+	cpuInt, _ := selector.Cpu.AsInt64()
+	ramInt, _ := selector.Memory.AsInt64()
+
```

Comment:
Fixed

Reply:
Fixed, as we discuss I also added other properties. I close this conversation so that we can have the last conversation to talk about other native metrics.